mpz-circuits/src/ops/binary.rs

```diff
@@ -212,19 +212,47 @@ pub(crate) fn inv_nbit<const N: usize>(
     std::array::from_fn(|n| state.add_inv_gate(a[n]))
 }
 
-impl<'a> BitXor<Tracer<'a, Bit>> for Tracer<'a, Bit> {
+impl<'a> BitXor for Tracer<'a, Bit> {
     type Output = Tracer<'a, Bit>;
 
     fn bitxor(self, rhs: Tracer<'a, Bit>) -> Self::Output {
-        let mut state = self.state.borrow_mut();
+        let out = self
+            .state
+            .borrow_mut()
+            .add_xor_gate(self.node(), rhs.node());
 
-        let out = state.add_xor_gate(self.to_inner().nodes()[0], rhs.to_inner().nodes()[0]);
+        Tracer::new(self.state, Bit::new([out]))
+    }
+}
+
+impl<'a> BitAnd for Tracer<'a, Bit> {
+    type Output = Tracer<'a, Bit>;
+
+    fn bitand(self, rhs: Self) -> Self::Output {
+        let out = self
+            .state
+            .borrow_mut()
+            .add_and_gate(self.node(), rhs.node());
+
+        Tracer::new(self.state, Bit::new([out]))
+    }
+}
+
+impl<'a> Not for Tracer<'a, Bit> {
+    type Output = Tracer<'a, Bit>;
 
-        let value = Bit::new([out]);
+    fn not(self) -> Self::Output {
+        let out = self.state.borrow_mut().add_inv_gate(self.node());
 
-        drop(state);
```

Comment:
The `RefMut` only lives to the end of the method and the state isn't used before then, so there's no need to explicitly drop it.